src/main/kotlin/com/chattriggers/ctjs/minecraft/wrappers/Client.kt

```diff
@@ -82,19 +85,39 @@ abstract class Client {
         //$$ getMinecraft().connection
         //#endif
 
+        /**
+         * Schedule's a task to run on Minecraft's main thread in [delay] ticks.
+         * Defaults to the next tick.
+         * @param delay The delay in ticks
+         * @param callback The task to run on the main thread
+         */
+        @JvmOverloads
+        @JvmStatic
+        fun scheduleTask(delay: Int = 0, callback: () -> Unit) {
+            ClientListener.tasks.add(Task(delay) {
+                if (getMinecraft().isCallingFromMinecraftThread) {
+                    callback()
+                } else {
+                    getMinecraft().addScheduledTask(callback)
```

Comment:
Is this if statement necessary? It looks like `Minecraft::addScheduledTask` already includes a check for `isCallingFromMinecraftThread` which leads to directly calling the `Runnable` if it is `true`.

Reply:
Also, if `Task` is moved to `ClientListener`, it can probably just hold `callback`, and `ClientListener::onTick` can handle the scheduling stuff

Reply:
> Also, if `Task` is moved to `ClientListener`, it can probably just hold `callback`, and `ClientListener::onTick` can handle the scheduling stuff

What do you mean by this? I had it so you could schedule something x ticks in the future, so how would that work without storing the delay with it?

Reply:
Sorry, I meant hold `callback` instead of the wrapping lambda. It'd still hold delay, so this code would just be `Task(delay, callback)`